include/swift/AST/ASTContext.h

```diff
@@ -869,6 +869,27 @@ class ASTContext final {
   /// This guarantees that resulted \p names doesn't have duplicated names.
   void getVisibleTopLevelModuleNames(SmallVectorImpl<Identifier> &names) const;
 
+  struct OverrideSignatureKey {
+    StringRef baseMethodSigString;
```

Comment:
Can you use the pointer values of the CanGenericSignatures instead of strings here?